wordpress-mobile/WordPress-iOS
# Migrate xmlrpc guess to nsurlsession

Refs #5245 

This PR migrate the code existing on WordPressApi pod, to guess the full URL of an XMLRPC endpoint for a self-hosted site, to a swift codebase and using NSURLSession for network code.

The end result is a Swift class called ```WordPressOrgXMLRPCValidator``` that reimplements all the code on WordPressApi method ```guessXMLRPCURLForSite```

All the associated unit tests where also migrated to Swift unit tests, to make sure all the edge cases are tested.

This PR does't start use the new class on the app yet, that will be done be a follow up PR that will remove the WordPressAPI pod from the project and update the code to use this new class.

To test:
  - Run the unit tests on ```WordPressOrgXMLRPCValidatorTests.swift```
  - Check if they all pass

Needs review: @astralbodies can you have a look on this?

@koke, if you are still around this week and have some free time, your opinion on this is welcome?
